### ISSUE TITLE
found: T extends @Initialized @Nullable Object. required: T extends @Initialized @Nullable Object.

### ISSUE THREAD

**Issue by pallosp** [typetools/checker-framework#783]
The nullness checker reports an odd error for the following class:

```
import javax.annotation.Nullable;

public class Holder<T> {
  @Nullable private T val;

  public void set(@Nullable T val) {
    this.val = val;
  }

  @Nullable
  public T get() {
    return val;
  }
}
```

<pre>javac -processor nullness Holder.java</pre>

<pre>
Holder.java:7: error: [assignment.type.incompatible] incompatible types in assignment.
    this.val = val;
               ^
  found   : T extends @Initialized @Nullable Object
  required: T extends @Initialized @Nullable Object
1 error</pre>

**Comment by pallosp** [comment]
I'm using version 2.0.0

**Comment by mernst** [comment]
I have confirmed this problem and added test cases in the following files:
```
   checker/tests/nullness/generics/Issue783a.java
   checker/tests/nullness/generics/Issue783b.java
   checker/tests/nullness/generics/Issue783c.java
```

Using -AprintVerboseGenerics indicates that the problem is with the `super` part of the type variable's type:
```
Issue783b.java:10: error: [assignment.type.incompatible] incompatible types in assignment.
    this.val = val;
               ^
  found   : T[ extends @org.checkerframework.checker.initialization.qual.Initialized @org.checkerframework.checker.nullness.qual.Nullable Object super @org.checkerframework.checker.initialization.qual.Initialized @org.checkerframework.checker.nullness.qual.Nullable Void]
  required: T[ extends @org.checkerframework.checker.initialization.qual.Initialized @org.checkerframework.checker.nullness.qual.Nullable Object super @org.checkerframework.checker.initialization.qual.Initialized @org.checkerframework.checker.nullness.qual.NonNull Void]
```

The error does not occur if I change 
```
import javax.annotation.Nullable;
```
to
```
import org.checkerframework.checker.nullness.qual.Nullable;
```
so perhaps the problem is related to annotation aliasing.  However, note that the annotations in the error message above are in org.checkerframework.checker.nullness.qual.

**Comment by smillst** [comment]
This is a duplicate of #471

**Comment by mernst** [comment]
Furthermore, #471 is now fixed, by commit 99b896a.
